capstone/src/main/java/com/google/sps/gmail/GmailAPILoader.java

```diff
@@ -0,0 +1,98 @@
+package com.google.sps.gmail;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.gmail.Gmail;
+import com.google.common.collect.Streams;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.security.GeneralSecurityException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.json.JSONObject;
+
+public class GmailAPILoader {
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final String APPLICATION_NAME = "clubhub-step-2020";
+
+  public static Gmail getGmailService() throws IOException, GeneralSecurityException {
+    // Add credentials for using API
+    Credential authorize =
+        new GoogleCredential.Builder()
+            .setTransport(GoogleNetHttpTransport.newTrustedTransport())
+            .setJsonFactory(JSON_FACTORY)
+            .setClientSecrets(Secrets.CLIENT_ID, Secrets.CLIENT_SECRET)
+            .build()
+            .setRefreshToken(Secrets.REFRESH_TOKEN)
+            .setAccessToken(getAccessToken());
+
+    // Create Gmail API Service
+    final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+    Gmail service =
+        new Gmail.Builder(HTTP_TRANSPORT, JSON_FACTORY, authorize)
+            .setApplicationName(APPLICATION_NAME)
+            .build();
+    return service;
+  }
+
+  private static String getAccessToken() throws IOException {
```

Comment:
This method does not need this since you already catch all exceptions.

Reply:
Removed the IOException, thank you for catching this!